samples/python/reactors/PorousMediaBurner.py

```diff
@@ -0,0 +1,395 @@
+based on extensible reactors. Showcases adding a temperature equation for a
+solid-phase and custom heat transfer/radiation models.
+
+Code by Thorsten Zirwes and Guillaume Vignat
+Stanford University & Karlsruhe Institute of Technology (KIT)
+2023
+
+This code implements a reactor cascade model for the simulation of reactive
+flows in porous media. The gas mixture with fuel and oxidizer flows through
+the porous material and the chemical reactions take place inside the pore
+cavities. The heat released by the chemical reactions heats up the solid materials.
+This heat is then transported by conduction in the solid and radiation upstream
+to heat up the reactants and stabilize the reaction zone.
+Given a reaction mechanism, equations for species mass
+fractions, gas phase temperature and solid temperature are solved.
+The reactor cascade consists of a variable number of cylindrical reactors.
+The Rosseland model is used for radiative heat transfer by the solid.
+A Nusselt number model is used to compute the heat transfer coefficient.
+This example assumes finding steady-state solutions with constant pressure
+so that mass controllers and pressure valves are not required. The code is written in
+a general way to support an arbitrary number of reactors and an arbirary number of
+burner sections with different physical properties.
+
+The implemented equations make use of Cantera's extensible reactor models.
+Therefore, Cantera version 2.6.0a4 or higher is required.
+
+The porous media burner considered in this example is a cylindrical tube filled
+three different porous materials: a porous ceramic made from a Yttria-stabilized
+Zirconia Alumina (YZA) section with length of 2 inches and pore density of 40 pores per
+inches (PPI), then a one inch section with a porous ceramic made from
+silicon carbide (SiC) with 3 PPI and finally a 1 inch section of SiC with 10 PPI. A
+flame stabilizes at the interface between the YZA and 3 PPI SiC.
+
+
+          |--- two inches --- | --- one inch ---| --- one inch --- |
+          __________________________________________________________
+fuel
+ +   =>      YZA 40 PPI       |     SiC 3 PPI   |   SiC 10 PPI       => burnt
+air       __________________________________________________________    gas
+
+                inert            flame location   heat recirculation
+
+
+More details about the governing equations, submodels and physical setup and
+corresponding experiments can be found in
+
+    Experimental and numerical investigation of flame stabilization and
+    pollutant formation in matrix stabilized ammonia-hydrogen combustion,
+    G. Vignat, T. Zirwes, E.R. Toro, K. Younes, E. Boigne, P. Muhunthan,
+    L. Simitz, D. Trimis, M. Ihme
+    Combustion and Flame (https://doi.org/10.1016/j.combustflame.2023.112642)
+
+Requires: cantera >= 2.6.0a4
+Keywords: porous media burner, heat transfer between gas and solid,
+          radiation modeling, reactor cascade
```

Comment:
The required version should just be "2.6.0". You should also list `matplotlib` as a dependency (we usually put ">= 2.0" as the version requirement).

For the keywords, please see the list in `doc/example-keywords.txt` and add any that seem relevant. "user-defined model" is the one that covers the use of `ExtensibleReactor`. If you want to add any beyond the existing ones, they should be ones where there's some likelihood of (eventually) having more examples that use the same keywords. I think I'd suggest:
```
Keywords: user-defined model, reactor network, combustion, porous media, heat transfer, radiative heat transfer
```